docs/source/RunningInEmulationTCG.rst

```diff
@@ -0,0 +1,92 @@
+.. _emulation:
+
+===============================
+Running in emulation mode (TCG)
+===============================
+
+Sometimes it's useful to run Avocado-vt in emulation mode (eg. when
+checking PR about different architecture, or to debug certain feature
+while executing the test). As this is not default, extra arguments are
+usually required.
+
+Shared for all architectures is that you need the image. We supply
+JeOS for most of the architectures, so you should be able to get
+it via::
+
+    $ avocado vt-bootstrap --vt-guest-os JeOS.$VERSION.$ARCH
+
+where:
+
+* ``VERSION`` is JeOS version (when writing this document it was ``27``)
+* ``ARCH`` is the desired architecture (eg. aarch64)
+
+Or you can run any of the ``unattended_install`` tests with
+``--vt-guest-os`` of your choice (very slow).
+
+When running the tests, on top of the usual arguments, you usually
+need to specify:
+
+* ``--vt-qemu-bin`` - path to binary that supports expected architecture
+  (eg. ``/usr/local/bin/qemu-system-aarch64``)
+* ``--vt-qemu-dst-bin`` - for migration tests you need also to specify
+  the destination qemu binary (otherwise default one is used)
+* ``enable_kvm=no`` - to disable KVM (if necessary)
+
+
+aarch64
+=======
+
+ARM always requires `cpu_model` as well as `machine_type`. To
+get list of available models you can run ``qemu-system-aarch64
+-cpu help -M virt`` (note: not all listed cpus are bootable).
+ARM also requires GIC version to be specified. Avocado-vt
+sets automatically the best via ``gic=host``, but that is not
+possible without KVM so you need to override the Avocado-vt
+default by overriding ``machine_type_extra_params``. Complete
+example would be::
+
+   $ avocado vt-bootstrap --vt-guest-os JeOS.27.aarch64
+   $ avocado --show all run --vt-extra-params enable_kvm=no cpu_model=cortex-a57 \\
+       machine_type_extra_params='' --vt-machine-type aarch64 --vt-arch arm64-pci \\
```

Comment:
I wonder why there are two "\" characters at the end of each command line? Shouldn't it be one instead? Is this a requirement of rst document?

Reply:
Definitely not one, bash would use them as a string. They are not even necessary, I just have them in my history as sometimes I need to specify things that require bash escapes and when having it empty I just keep them for future changes... Also IMO it looks cleaner to mark I want an empty string, but I'm not attached to it.

Reply:
... I mean they are necessary when one wants to insert `something='foo bar baz'`, or `something=(foo)` or other things that would be escaped by bash.

Reply:
Oh, good catch. Thank you.